python/fixedlen_tagger.py

```diff
@@ -38,7 +38,10 @@ def __init__(self, syncword_tag, packetlen_tag, packet_len, stream_type):
 
     def set_packet_len(self, packet_len):
         self.packet_len = packet_len
-        self.stream.maxlen = self.packet_len
+        copy = self.stream.copy()
+        self.stream = collections.deque(maxlen=self.packet_len - 1)
+        for entry in copy:
+            self.stream.append(entry)
```

Comment:
Thanks! I think this would work, but perhaps it can be made more efficient like so:
```
self.stream = collections.deque(self.stream, maxlen=self.packet_len-1)
```
The copy doesn't seem necessary (we could `old_stream = self.stream`), and instead of the `for` loop it's better to use the `.extend()` method of the deque. But actually the new deque can be constructed and initialized with an existing iterable, so it's not even necessary to use `old_stream` or `.extend()`.

Reply:
Yup, that looks much better. I believe it works just as well. Pushed another fix!